lib/stages/visTranspose.cpp

```diff
@@ -192,17 +192,23 @@ bool visTranspose::get_dataset_state(dset_id_t ds_id) {
     write_f = chunk_f;
 
     // Allocate memory for collecting frames
-    vis.resize(chunk_t * chunk_f * eff_prod_dim);
-    vis_weight.resize(chunk_t * chunk_f * eff_prod_dim);
-    eval.resize(chunk_t * chunk_f * num_ev);
-    evec.resize(chunk_t * chunk_f * num_ev * num_input);
-    erms.resize(chunk_t * chunk_f);
-    gain.resize(chunk_t * chunk_f * num_input);
-    frac_lost.resize(chunk_t * chunk_f);
-    frac_rfi.resize(chunk_t * chunk_f);
+    vis.resize(chunk_t * chunk_f * eff_prod_dim, 0.);
+    vis_weight.resize(chunk_t * chunk_f * eff_prod_dim, 0.);
+    eval.resize(chunk_t * chunk_f * num_ev, 0.);
+    evec.resize(chunk_t * chunk_f * num_ev * num_input, 0.);
+    erms.resize(chunk_t * chunk_f, 0.);
+    gain.resize(chunk_t * chunk_f * num_input, 0.);
+    // init frac_lost to 1.0 to match empty frames
+    frac_lost.resize(chunk_t * chunk_f, 1.);
+    frac_rfi.resize(chunk_t * chunk_f, 0.);
+    input_flags.resize(chunk_t * num_input, 0.);
     dset_id.resize(chunk_t * chunk_f);
-    input_flags.resize(chunk_t * num_input);
-    std::fill(input_flags.begin(), input_flags.end(), 0.);
+
+    // Initialise dataset ID array with null IDs
```

Comment:
I wonder if we can get the same result with a simpler initialization value. How often does `chunk_t` / `num_time` change, triggering an actual resize?

Reply:
it only happens once at the start for a given file. The reason it needs to be initialized explicitly now is that the empty frames at the start of the file will not be copied into these buffer arrays so they need to start out with sane values. After that, empty frames are copied just like filled frames